### PR TITLE
Add connectivity handler

### PR DESCRIPTION
**Pull request by ancaantochi**
Adding handler for connectivity events. 
The idea is to create the channel in the bridge and the sender is passed to upstream pump and receiver to downstream pump. This will be in a different PR since changes are based on the pump.

Note: PR has to be merged after changes to mqtt3 client changes.

### CODE REVIEW

File: mqtt/mqtt-bridge/src/bridge.rs [hunk 1]
```diff
@@ -1,8 +1,11 @@
+#![allow(dead_code)] // TODO remove when ready
+
 use std::{collections::HashMap, convert::TryFrom, convert::TryInto};
 
 use async_trait::async_trait;
 use mqtt3::{proto::Publication, Event, ReceivedPublication};
 use mqtt_broker::TopicFilter;
+use tokio::sync::mpsc::error::SendError;
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-bridge/src/bridge.rs:8]: nit: rearrange

File: mqtt/mqtt-bridge/src/connectivity_handler.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
+#![allow(dead_code)] // TODO remove when ready
+
+use mqtt3::Event;
+use tokio::sync::mpsc::UnboundedSender;
+use tracing::{debug, info, warn};
+
+use crate::bridge::{BridgeError, BridgeMessage, ConnectivityState};
+
+impl ToString for ConnectivityState {
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-bridge/src/connectivity_handler.rs:9]: it's better to put trait bloks after impl. In general recommended patten is:
- struct
- impl for struct
- impl trait for struct
- free fn

File: mqtt/mqtt-bridge/src/connectivity_handler.rs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+impl ConnectivityHandler {
+    pub fn new(sender: UnboundedSender<BridgeMessage>) -> Self {
+        ConnectivityHandler {
+            state: ConnectivityState::Disconnected,
+            sender,
+        }
+    }
+
+    pub async fn handle_connectivity_event(&mut self, event: Event) -> Result<(), BridgeError> {
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-bridge/src/connectivity_handler.rs:32]:
```suggestion
    pub fn handle(&mut self, event: Event) -> Result<(), BridgeError> {
```
does it have to be async?

Comment (ancaantochi) [reply mqtt/mqtt-bridge/src/connectivity_handler.rs]: I changed to impl EventHandler which is async

File: mqtt/mqtt-bridge/src/lib.rs [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
 mod bridge;
 pub mod client;
+mod connectivity_handler;
```
Comment (dmolokanov) [inline mqtt/mqtt-bridge/src/lib.rs:16]:
```suggestion
mod connectivity;
```
?

File: mqtt/mqtt-bridge/src/bridge.rs [hunk 5]
```diff
@@ -15,6 +24,43 @@ use crate::{
...
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ConnectivityState {
+    Connected,
+    Disconnected,
+}
+
+impl Display for ConnectivityState {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        match *self {
+            ConnectivityState::Connected => f.write_str("Connected"),
```
[the remark this answers is not in the document]
Comment (ancaantochi) [reply mqtt/mqtt-bridge/src/bridge.rs]: I think I took it from some other place, changed to write!

File: mqtt/mqtt-bridge/src/bridge.rs [hunk 6]
```diff
@@ -15,6 +24,43 @@ use crate::{
...
+}
+
+impl PumpHandle {
+    pub fn new(sender: Sender<PumpMessage>) -> Self {
+        Self { sender }
+    }
+
+    pub fn send(&mut self, message: PumpMessage) -> Result<(), BridgeError> {
+        self.sender
+            .try_send(message)
```
[the remark this answers is not in the document]
Comment (ancaantochi) [reply mqtt/mqtt-bridge/src/bridge.rs]: actually I think it's not correct to use try_send, I changed to send

File: mqtt/mqtt-bridge/src/connectivity_handler.rs [hunk 7]
```diff
@@ -0,0 +1,205 @@
+#![allow(dead_code)] // TODO remove when ready
+
+use async_trait::async_trait;
+
```
[the remark this answers is not in the document]
Comment (ancaantochi) [reply mqtt/mqtt-bridge/src/connectivity_handler.rs]: I rearranged, is there a way to get this automatically?

**Review by vadim-kovalyov — DISMISSED** [review]
:rocket:

**Review by vadim-kovalyov — APPROVED** [review]

**Pull request merged**